adage/genes/views.py

```diff
@@ -1,3 +1,57 @@
-from django.shortcuts import render
+from rest_framework.viewsets import ReadOnlyModelViewSet
+from .models import Gene
+from .serializers import GeneSerializer
 
-# Create your views here.
+
+class GeneViewSet(ReadOnlyModelViewSet):
+    """Genes viewset"""
+
+    http_method_names = ['get']
+    serializer_class = GeneSerializer
+
+    def get_queryset(self):
+        queryset = Gene.objects.all()
+
+        # 'search' parameter, which does full text search based on the
+        # following four fields in Gene model:
+        # - "standard_name": highest priority (A: 1.0);
+        # - "systematic_name": second highest priority (B: 0.4);
+        # - "description" and "crossref__xrid": lowest priority (C: 0.2).
```

Comment:
See my note in `analyze/views.py` about making the first part of this comment less terse.

In this case, I am also curious about how you chose the weighting factors. Why are `standard_name` and `systematic_name` weighted differently, for example? I think it makes sense that matching the gene name should be stronger than matching the description but I wonder if a weight of 0.2 is too small for the description. Maybe we need a strategy for testing this with real queries that our users use. What do you think?

Reply:
The weights are chosen (kind of) arbitrarily. @cgreene and I had some discussion in issue #7. It's also possible to set our own weights instead of using the default A/B/C/D.

Reply:
As you said, we can (and probably should) ask our users to test the search results and tweak the weights based on their feedback.

Reply:
Hmm. I missed the discussion at #7 even though you referenced my GitHub handle. I need to look into why I'm not getting these notifications.

Thanks, I'll follow up there.